src/libbson/src/bson/bson-context.c

```diff
@@ -264,111 +107,242 @@ _bson_context_get_hostname (char *out)
 }
 
 
-static void
-_bson_context_init_random (bson_context_t *context, bool init_sequence)
+/*** ========================================
+ * The below SipHash implementation is based on the original public-domain
```

Comment:
Comment seems misplaced; should be moved to immediately before the `_siphash()` function definition.

Reply:
Everything between this comment and the end of `_siphash` are all based on the original impl. I'd like to reverse their decl order, but we're in C, so they must be declared bottom-up.